### ISSUE TITLE
Gas mining mission not working properly, and a lot of lag

### ISSUE THREAD

**Issue by jstaff62** [zmaster587/AdvancedRocketry#914]
Minecraft Version:
- 1.7.10
Forge Version:
- Not sure (it's the latest)
LibVulpes Version:
- 0.2.6
*is this the latest curseforge build?*
Yes
**Please Pastebin crash**
N/A
Crash Report:
N/A
Explain issue in detail:

My gas mining rockets are set up properly, all 4 of them. I can launch them fine and i do so automatically and they return fine, but there are two big problems: Firstly when they take off, they disappear when they leave render distance so the mission doesn't actually start. The monitoring stations read "mission progress: N/A" and incorrectly shows the rockets are full of fuel. however, flying into the chunks that they disappeared in causes them to reappear mid-flight and the mission starts and continues as usual. this makes automation impossible.

Another issue is that the game turns into a slideshow whenever the rockets are formed, but docked. as soon as they take off, all lag is instantly gone as the last one leaves, also when they're disassembled. (I usually get around 60-80 fps, but this takes it down to around 10 fps)

**Comment by gdarai** [comment]
Other mods? What is your configuration? Did you tried it also on Vanilla + AdvancedRocketry only?

**Comment by zmaster587** [comment]
Can you get a profile of the lag issue with visualvm or simiar.  It's actually rather odd because the only difference between docked and not is velocity, all the same stuff gets rendered and run.  Also I typically get around 20 to 30fps in vanilla and have not encountered this while testing.  As for the issue with it not being automatable, can you use chicken chunks or similar to check to see if the rocket is correctly registering as a chunkloader?

**Comment by jstaff62** [comment]
I'll look into that visualvm and blank world testing and get back to you, but from just running them a bit more, I think the lag mainly pops up when the rockets are docked and empty, ready to be launched, and then, mentioned above the lag is completely gone as soon as they're launched. I have them docked and full of fluid with no lag. Also I assume you mean just any chunkloader? I had to uninstall chicken chunks because those loaders were constantly causing crashes. The entire station and a large around it are all loaded with a rotarycraft chunkloader.

**Comment by zmaster587** [comment]
The whole lag issue still baffles me, I am curious to see the results of a visualvm profile.

The autonomous rockets themselves are supposed to be a chunkloader, so they can make it out far enough to begin the mission (and still return if the player happens to move around a bit).  Chicken chunks has a utility to show a map of loaded chunks.  It doesn't have to be chicken chunk, just some way of knowing if the rocket is actually loading the chunks like it should be.

**Comment by jstaff62** [comment]
Just tested, it's a mod interaction issue i think. everything works fine with just AR and chicken chunks in a blank world, and i didn't even need to try the chunk viewer thing, it just worked. so it seems i'm either gonna have to put up with it or have fun testing combinations of around 120 mods to find the problematic one...

**Comment by jstaff62** [comment]
I don't think providing a visualvm profile is going to be possible now after trying it and looking at tutorials, I can't even run it as I have no clue what I'm doing with any of these files or commands.

**Comment by jstaff62** [comment]
Just an update, the lag issue looks to be an issue with how the fluid unloader reacts with other mods. the block is only set to send a signal out of one side, but having an ender io redstone conduit, even unconnected, causes lag, so does a TE fluiduct, even without a servo for redstone control. removing all of this fixes the lag. basically it seems the unloader in at least some cases doesn't like having any redstone entities around it, connected or not, while it's sending a signal, which it was in my case, being docked docked and empty. this obviously wasn't a problem in the blank test pack.

**Comment by zmaster587** [comment]
the fact that they only need to be nearby and not actually connected is baffling to me...  As for running visualvm there shouldn't be many commands you need to run.  You should only need to double click the minecraft java instance from the main menu on the left side of the screen, then go to the sampler tab, then click CPU.  From there just focus the game window for a few seconds (it behaves a bit differently when not focused).  then in visualvm again, hit stop then snapshot.  After clicking snapshot you should see a bunch of bars in the workspace with names like thread-0 etc, at that point, just click the export to button in the window and upload the resultant file.  It should contain exactly why it's lagging and would make my life easier I hope.  In theory.

I hadn't considered the max size of the station being a problem, I did double check and verify only players are affected.  Rockets and other entities won't be.  however there is an interesting edge case.  If the rocket does go outside of station boundries before "descending to the planet" it may get confused as to the planet it's going to.  What I'd expect to happen is that it would just disappear and never come back or maybe come back empty.  but neither of those fit the issue you're experiencing and definatly wouldn't prevent the mission from starting.

With the most recent update it still sounds like it could be a chunkloading issue.  Do you ever see a line in the server logs indicating advanced rocketry tried to allocate a chunk loader and was unable to?

**Comment by jstaff62** [comment]
Just tested in the blank world with the rockets a fair distance away from the center, they worked okay, even though the ones in my main world in about the same position didn't. before this though, I was playing, had a crash (just something random about reactorcraft fusion plasma and CPW event handler), but when I came back on I had 1800ms tick lag coming from the space station. I visited to see what was going on, looked at the rocket coming back and it kept bouncing up and down in its frame, just being generally really glitchy, trying to scan/rebuild and it had no clue what it was doing.

Also my problem with visualvm was that I couldn't open it because it was trying to use something to run it and was asking for jdk 8? which i think I installed but the whole thing is confusing me and I don't want to mess anything up. I wonder if all of this could be explained by corruption of some sort, though I have no idea how it could have happened, or even if it did happen and what it could have affected. I'll just attach a log here if that's okay. Last time I tried pastebin it was far too long and wouldn't work.

Here it is:
[2017-11-04-3.log.gz](https://github.com/zmaster587/AdvancedRocketry/files/1443602/2017-11-04-3.log.gz)


sorry if I'm making this all really complicated haha, I'm just a musiciscian who doesn't know the first thing about programming so things that might seem obvious to you and others just go completely over my head
